internal/provider/data_source_account.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/boundary/api"
+	"github.com/hashicorp/boundary/api/accounts"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAccount() *schema.Resource {
+	return &schema.Resource{
+		Description: "The boundary_account data source allows you to find a Boundary account.",
+		ReadContext: dataSourceAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the retrieved account.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description:  "The name of the account to retrieve.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			DescriptionKey: {
+				Description: "The description of the retrieved account.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			AuthMethodIdKey: {
+				Description:  "The auth method ID that will be queried for the account.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func dataSourceAccountRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	md := meta.(*metaData)
+	opts := []accounts.Option{}
+
+	var name string
+	if v, ok := d.GetOk(NameKey); ok {
+		name = v.(string)
+	} else {
+		return diag.Errorf("no name provided")
+	}
+
+	var authMethodId string
+	if authMethodIdVal, ok := d.GetOk(AuthMethodIdKey); ok {
+		authMethodId = authMethodIdVal.(string)
+	} else {
+		return diag.Errorf("no auth method ID provided")
+	}
+
+	acl := accounts.NewClient(md.client)
+
+	als, err := acl.List(ctx, authMethodId, opts...)
+	if err != nil {
+		return diag.Errorf("error calling list account: %v", err)
+	}
+	if als == nil {
+		return diag.Errorf("no accounts found")
+	}
+
+	var accountIdRead string
+	for _, accountItem := range als.GetItems() {
+		if accountItem.Name == name {
+			accountIdRead = accountItem.Id
```

Comment:
So in this case if there are multiple accounts associated to the auth method, we only return the first one? 
I am asking because in [boundary.data_source_user](https://github.com/hashicorp/terraform-provider-boundary/blob/786dfc11cc76be4d3d17134b17b78addb6fcd357/internal/provider/data_source_user.go#L124), if there are multiple users, we throw an error.
We should possibly standardized how we handle multiple results for datasources unless there's a good reason to differ

Reply:
I *think* there are checks where resources can't have the same name? 

I do agree we should try to standardize how we implement these. I have noticed some differences between `boundary_user` and `boundary_scope`.

Reply:
yeah that's true actually. There can only be 1 unique account name. ignore my comment then

Reply:
@elimt Made some changes here. Let me know what you think. Would like to apply the same paradigm to the other data sources.